api/models.py

```diff
@@ -50,3 +67,36 @@ def __str__(self):
 
     def __repr__(self):
         return f"{self.username}"
+
+
+@event.listens_for(Session, "before_flush")
+def session_before_flush(session, flush_context, instanses):
```

Comment:
okay, so this logging stuff seem pretty generic that you and Tolya use it. then it's a perfect use case to abstract it out to a library and specify as a project dependency.